tests/Microsoft.Identity.Test.Unit/TelemetryTests/RegionalTelemetryTests.cs

```diff
@@ -43,48 +65,39 @@ public class RegionalTelemetryTests : TestBase
-            {
-                Environment.SetEnvironmentVariable(TestConstants.RegionName, TestConstants.Region);
-
-                _app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
-                                                           .WithAuthority(AzureCloudInstance.AzurePublic, TestConstants.TenantId)
-                                                           .WithClientSecret(TestConstants.ClientSecret)
-                                                           .WithHttpManager(_harness.HttpManager)
-                                                           .WithExperimentalFeatures(true)
-                                                           .BuildConcrete();
-
-                Trace.WriteLine("Step 1. Acquire Token For Client with region successful");
-                var result = await RunAcquireTokenForClientAsync(AcquireTokenForClientOutcome.Success).ConfigureAwait(false);
-                AssertCurrentTelemetry(result.HttpRequest, ApiIds.AcquireTokenForClient, forceRefresh: false);
-                AssertPreviousTelemetry(result.HttpRequest, expectedSilentCount: 0); // Previous_request = 2|0|||
-
-                Trace.WriteLine("Step 2. Acquire Token For Client -> HTTP 5xx error (i.e. AAD is down)");
-                result = await RunAcquireTokenForClientAsync(AcquireTokenForClientOutcome.AADUnavailableError).ConfigureAwait(false);
-                Guid step2CorrelationId = result.Correlationid;
-
-                // we can assert telemetry here, as it will be sent to AAD. However, AAD is down, so it will not record it.
-                AssertCurrentTelemetry(result.HttpRequest, ApiIds.AcquireTokenForClient, forceRefresh: true);
-                AssertPreviousTelemetry(
-                    result.HttpRequest,
-                    expectedSilentCount: 0);
-
-                // the 5xx error puts MSAL in a throttling state, so "wait" until this clears
-                _harness.ServiceBundle.ThrottlingManager.SimulateTimePassing(
-                    HttpStatusProvider.s_throttleDuration.Add(TimeSpan.FromSeconds(1)));
-
-                Trace.WriteLine("Step 3. Acquire Token For Client -> Success");
-                result = await RunAcquireTokenForClientAsync(AcquireTokenForClientOutcome.Success, true).ConfigureAwait(false);
-
-                AssertCurrentTelemetry(result.HttpRequest, ApiIds.AcquireTokenForClient, forceRefresh: true);
-                AssertPreviousTelemetry(
-                    result.HttpRequest,
-                    expectedSilentCount: 0,
-                    expectedFailedApiIds: new[] { ApiIds.AcquireTokenForClient },
-                    expectedCorrelationIds: new[] { step2CorrelationId },
-                    expectedErrors: new[] { "service_not_available" },
-                    expectedRegions: new[] { "centralus" });
-            }
+            Environment.SetEnvironmentVariable(TestConstants.RegionName, TestConstants.Region);
```

Comment:
This enviroment variable is there to stay, i.e. this test will modify the global state of the machine. This will create problems when other regional tests run, because you do not have control over the order of the tests. 

Best practice is to cleanup or reset the global state. I would do both: 

1. All regional tests that set this env variable should have a "finally" clause where you clear the env variable (or you can do it via [TestCleanup])
2. All regional tests should clear this enviroment variable as part of their [TestSetup]

Reply:
Done